test/test_autograd.py

```diff
@@ -7016,6 +7026,249 @@ def test_size_check(self):
 
             dual = fwAD.make_dual(foo, tangent[1:])
 
+    # The following test functions want to ensure all the following behaviors:
+    #   - Ensure that default level system in the python binding works
+    #   - Ensure that only level 0 exists and nesting is properly disabled
+    #   - Ensure that printing works fine
+    #   - Ensure that basic packing/unpacking works
+    #   - Ensure that advanced packing/unpacking works
+    #     - For memory / version counter share
+    #     - For backward AD (regular ops)
+    #   - Ensure that view + inplace for both modes work fine
+    #   - Ensure we do proper cleanup on exit of a level
+
```

Comment:
Suggestion for testing: there's not a lot of code that tests that the codegen works (though I do see things further up the stack that would verify correctness). Unittesting some of the load_derivatives code (e.g., replacing variables in the formulas) and testing some malformed input gives nice error messages would help solve the chicken-and-egg problem that we're currently in (that it's hard to test the codegen without writing a bunch of formulas)